pkg/webhook/resources/virtualmachinebackup/validator.go

```diff
@@ -177,3 +190,66 @@ func (v *virtualMachineBackupValidator) Delete(_ *types.Request, obj runtime.Obj
 	}
 	return nil
 }
+
+func (v *virtualMachineBackupValidator) checkTotalSnapshotSize(vm *kubevirtv1.VirtualMachine) error {
+	resourceQuota, err := v.resourceQuotaCache.Get(vm.Namespace, util.DefaultResourceQuotaName)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil
+		}
+		return werror.NewInternalError(fmt.Sprintf("failed to get resource quota %s/%s, err: %s", vm.Namespace, util.DefaultResourceQuotaName, err))
+	}
+
+	if resourceQuota.Spec.SnapshotLimit.VMTotalSnapshotSizeQuota != nil {
+		if err := v.checkTotalSnapshotSizeOnVM(vm, resourceQuota.Spec.SnapshotLimit.VMTotalSnapshotSizeQuota[vm.Name]); err != nil {
+			return err
+		}
+	}
+
+	if err := webhookutil.CheckTotalSnapshotSizeOnNamespace(v.pvcCache, v.engineCache, vm.Namespace, resourceQuota.Spec.SnapshotLimit.NamespaceTotalSnapshotSizeQuota); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (v *virtualMachineBackupValidator) checkTotalSnapshotSizeOnVM(vm *kubevirtv1.VirtualMachine, totalSnapshotSizeQuota int64) error {
+	if totalSnapshotSizeQuota == 0 {
+		return nil
+	}
+
+	var totalSnapshotUsage int64
+	for _, volume := range vm.Spec.Template.Spec.Volumes {
+		if volume.PersistentVolumeClaim == nil {
+			continue
+		}
+
+		pvcNamespace := vm.Namespace
+		pvcName := volume.PersistentVolumeClaim.ClaimName
+
+		pvc, err := v.pvcCache.Get(pvcNamespace, pvcName)
+		if err != nil {
+			return werror.NewInternalError(fmt.Sprintf("failed to get PVC %s/%s, err: %s", pvcNamespace, pvcName, err))
+		}
+
+		engine, err := webhookutil.GetLHEngine(v.engineCache, pvc.Spec.VolumeName)
+		if err != nil {
+			return werror.NewInternalError(fmt.Sprintf("failed to get engine for PVC %s/%s, err: %s", pvcNamespace, pvcName, err))
+		}
+		for _, snapshot := range engine.Status.Snapshots {
+			if snapshot.Removed {
+				continue
+			}
+
+			snapshotSize, err := longhornutil.ConvertSize(snapshot.Size)
+			if err != nil {
+				return werror.NewInternalError(fmt.Sprintf("failed to convert snapshot size %s to bytes, err: %s", snapshot.Size, err))
+			}
+			totalSnapshotUsage += snapshotSize
+		}
+	}
+
+	if totalSnapshotSizeQuota < totalSnapshotUsage {
+		return werror.NewInvalidError(fmt.Sprintf("total snapshot size limit %d is less than total snapshot usage %d", totalSnapshotSizeQuota, totalSnapshotUsage), "spec.source.name")
```

Comment:
It would nice to show current usage and total snapshot quota in Gi.

Current UI only allow user to input quota in GB.
<img width="1467" alt="Screenshot 2024-08-21 at 3 31 12 PM" src="https://github.com/user-attachments/assets/f0026c96-a88b-4483-b46b-6b2aba4d310b">

Reply:
Updated it.
<img width="587" alt="Screenshot 2024-08-23 at 12 30 49 PM" src="https://github.com/user-attachments/assets/c9e721d5-9db2-4fb0-934e-b05746ff9b27">